video/out/wayland_common.c

```diff
@@ -1370,6 +1376,16 @@ static void main_device(void *data,
                         struct zwp_linux_dmabuf_feedback_v1 *zwp_linux_dmabuf_feedback_v1,
                         struct wl_array *device)
 {
+    struct vo_wayland_state *wl = data;
+
+    // Despite being an array, the protocol specifically states there can only be
+    // one main device so break as soon as we get one.
+    dev_t *id;
+    wl_array_for_each(id, device) {
+        memcpy(&wl->main_device_id, id, sizeof(dev_t));
+        break;
+    }
+    get_gpu_drm_formats(wl);
```

Comment:
I think this should be under `tranche_target_device`. The protocol says: `This event advertises the target device that the server prefers to use for a buffer created given this tranche`

While for main_device, it says `This event advertises the main device that the server prefers to use when direct scan-out to the target device isn't possible`

Reply:
This wording seems suspicious. If you have only one device that can direct scanout, surely the main device is the one that is being used for direct scanout? But it's probably more conceptually correct to use the first `tranche_target_device` we get and ignore the rest within a sequence of main device -> tranche -> done events.

Reply:
You can have multiple devices that can direct scan out in multi gpu scenarios no? Suppose my compositor is running on gpu 0 but my mpv is on a screen connected to gpu 1. Our main device is gpu 0 here, but the target device is gpu 1 since we're not on a screen that's on gpu 0. I can ask for clarification here if you want

Reply:
Sure. But my expectation would be that the main device isn't necessarily the `device that the server prefers to use when direct scan-out to the target device isn't possible`. If you have only one device, the main device and the target have to be the same. It just depends on whatever your hardware is.

It doesn't really matter for us though.